frontend/model/contracts/shared/functions.js

```diff
@@ -265,3 +261,43 @@ export function swapMentionIDForDisplayname (
     .map(t => regEx.test(t) ? swap(t) : t)
     .join('')
 }
+
+export const referenceTally = (selector: string): Object => {
```

Comment:
Could you add a comment here explaining here what this is?

Reply:
Sure. Preliminarily, so that it doesn't block your review, this is a temporary / ephemeral reference count that is kept in memory until the 'real' release or retain operation is called.

So, if you join / leave / join / leave, and all of those actions are processed at once, this ensures that the reference count isn't affected, as those operations cancel out.

This is similar to what we were doing previously, except that this works and the other thing didn't. Before, the count managed by side-effects, which tried to guess based on the state.

The main difference is that, before, side effects pushed to the end of the queue each. Now, a single function is pushed to the end of the queue, no matter the number of actions.

Reply:
Comment added.